nano/connection/dimensional_to_dimensional.py

```diff
@@ -0,0 +1,207 @@
+from abc import ABCMeta
+
+import numpy as np
+
+from nano.connection import AbsConnection
+from nano.layer import DimensionalLayer
+from nano.layer import LinearLayer
+
+class DimensionToDimension(AbsConnection, metaclass=ABCMeta):
+	def __init__(self):
+		super().__init__()
+
+	def check_layer_type(self, before_layer, after_layer):
+		return isinstance(before_layer, DimensionalLayer) and isinstance(after_layer, DimensionalLayer)
+
+class DimensionToLinear(AbsConnection, metaclass=ABCMeta):
+	def __init__(self):
+		super().__init__()
+
+	def check_layer_type(self, before_layer, after_layer):
+		return isinstance(before_layer, DimensionalLayer) and isinstance(after_layer, LinearLayer)
+
+class Convolution(DimensionToDimension):
+	def __init__(self, height, width, stride=1):
+		super().__init__()
+		self.stride = stride
+		self.height = height
+		self.width = width
+
+	def check_shape(self, before_shape, after_shape):
+		if ((before_shape[1] - self.height) // self.stride + 1 == after_shape[1]
+			and (before_shape[2] - self.width) // self.stride + 1 == after_shape[2]):
+			return True
+		else:
+			return False
+
+	def prepare_connection(self, before_shape, after_shape):
+		self.depth = before_shape[0]
+		self.filters = after_shape[0]
+		for i in range(0, self.filters):
+			self.weight.append(# filters
+				2 * np.random.random((self.depth, self.height, self.width)) - 1 
+			)
+		self.weight.append(# weights
+			np.random.random(self.filters)
+		)
+
+	def forward(self):
+		for i in range(0, self.filters):
+			filter_vec = self.weight[i].flatten()
+			bias = np.array([self.weight[self.filters][i]])
+
+			calc_matrix = np.zeros((self.height * self.width * self.depth , 1))
+			for j in range(0, self.after_layer.shape[1]):
+				for k in range(0, self.after_layer.shape[2]):
+					temp = np.zeros((1, 1))
+					for d in range(0, self.depth):
+						layer_values = self.before_layer.result[d, j * self.stride : j * self.stride + self.height, k * self.stride : k * self.stride + self.width]
+						layer_values = layer_values.reshape((self.height * self.width , 1))
+						temp = np.vstack((temp, layer_values))
```

Comment:
`vstack` and `hstack` inside for loop is very inefficient. It reallocates and copies the data in every loop. Check `nano.trainer.rms` for preallocation example.